src/NServiceBus.Persistence.AzureStorage/Subscriptions/AzureSubscriptionStorage.cs

```diff
@@ -142,5 +190,11 @@ public int GetHashCode(Subscriber obj)
                 return (obj.Endpoint ?? string.Empty).Length;
             }
         }
+
+        internal class CacheItem
+        {
+            public DateTime Stored;
+            public Task<IEnumerable<Subscriber>> Subscribers;
```

Comment:
so i used this approach for several reasons.

 * Avoids an async state machine for `GetSubscriberAddressesForMessage`
 * Allows us to always return from this method in a non blocking way
 * defers the decision to await/or not to the caller
 * simplifies the code